src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -318,23 +319,46 @@
     <V> Optional<V> getProperty(Property<V> property);
 
     /**
-     * Query this inventory for inventories matching any of the supplied
-     * queries. Logical <code>OR</code> is applied between operands.
+     * Query this inventory with given {@link Query}
      *
-     * @param operations queries to check against
-     * @return the query result
+     * @param query The query
+     *
+     * @return The queried inventory
+     */
+    Inventory query(Query query);
+
+    /**
+     * Query this inventory with given {@link QueryType} and {@code param}.
+     *
+     * @param type The query type
+     * @param param The query parameter
+     * @param <P> The query parameter type
+     *
+     * @return The queried inventory
+     */
+    <P> Inventory query(QueryType.OneParam<P> type, P param);
```

Comment:
IMP this method should be removed as it isn't any better readable than query(type.of(param))